dissect/target/filesystems/config.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+import io
+import textwrap
+from logging import getLogger
+from typing import Any, BinaryIO, Iterator, Optional, Union
+
+from dissect.target import Target
+from dissect.target.exceptions import ConfigurationParsingError, FileNotFoundError
+from dissect.target.filesystem import Filesystem, FilesystemEntry, VirtualFilesystem
+from dissect.target.helpers import fsutil
+from dissect.target.helpers.configparser import ConfigurationParser, parse
+
+log = getLogger(__name__)
+
+
+class ConfigurationFilesystem(VirtualFilesystem):
+    __fstype__: str = "META:configuration"
+
+    def __init__(self, target: Target, path: str, **kwargs):
+        super().__init__(**kwargs)
+        self.root.top = target.fs.get(path)
+
+    def _get_till_file(self, path: str, relentry: FilesystemEntry) -> tuple[list[str], FilesystemEntry]:
+        entry = relentry or self.root
+
+        path = fsutil.normalize(path, alt_separator=self.alt_separator).strip("/")
+
+        if not path:
+            return [], entry
+
+        parts = path.split("/")
+
+        for idx, part in enumerate(parts):
+            # Resolve link
+            if entry.is_symlink():
+                entry = entry.readlink_ext()
+
+            if part == ".":
+                continue
+            elif part == "..":
+                entry = entry.up or self.root
+                continue
+
+            if entry is self.root:
+                entry = self.root.top
+
+            entry = entry.get(part)
+            if entry.is_file():
+                break
+
+        return parts[idx:], entry
+
+    def get(
+        self,
+        path: str,
+        relentry: Optional[FilesystemEntry] = None,
+        hint: Optional[str] = None,
+        collapse: Optional[Union[bool, set]] = None,
+        seperator: Optional[tuple[str]] = None,
+        comment_prefixes: Optional[tuple[str]] = None,
+    ) -> Union[FilesystemEntry, ConfigurationEntry]:
+        parts, entry = self._get_till_file(path, relentry)
+
+        for part in parts:
+            if isinstance(entry, ConfigurationEntry):
+                entry = entry.get(part)
+            else:
+                try:
+                    config_parser = parse(entry, hint, collapse, seperator, comment_prefixes)
+                    entry = ConfigurationEntry(self, entry.path, entry, config_parser)
```

Comment:
```suggestion
                    entry = ConfigurationEntry(self, entry.path, entry, config_parser)
```
the part should be added to `entry.path` to create the full path for the new `ConfigurationEntry`. Preferably with the `fsutil.join()`

Reply:
Ah, I missed that the first part in `parts` was the filename. Maybe add a comment about this before the `for`-loop. (Or add a doc string to `_get_till_file()` ;-)